sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java

```diff
@@ -158,12 +161,17 @@ public String serializeList(List<?> list, CollectionFormat format) {
     @Override
     @SuppressWarnings("unchecked")
     public <T> T deserialize(String value, final Type type, SerializerEncoding encoding) throws IOException {
-        if (value == null || value.isEmpty() || value.equals(BOM)) {
+        if (CoreUtils.isNullOrEmpty(value) || value.equals(BOM) || value.equals(BOM_STRING)) {
             return null;
         }
-        // Remove BOM
-        if (value.startsWith(BOM)) {
-            value = value.replaceFirst(BOM, "");
+
+        // Remove any leading BOM from the XML.
+        if (encoding == SerializerEncoding.XML) {
+            if (value.startsWith(BOM)) {
+                value = value.replaceFirst(BOM, "");
+            } else if (value.startsWith(BOM_STRING)) {
+                value = value.replaceFirst(BOM_STRING, "");
+            }
```

Comment:
As discussed, please file an issue for further investigation on this. This is good for now to release this as this issue was already noticed in smoke tests and would be better to get this fix out before customers are impacted.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/8710